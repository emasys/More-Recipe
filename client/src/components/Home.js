import React, { Component } from 'react';
import Catalog from '../containers/catalog';
import { Link } from 'react-router-dom';
import { getRecipes, searchRecipes } from '../actions';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

//component
import CatalogList from '../components/CatalogList';
import Navbar from './Navbar';

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      search: '',
      All_recipes: '',
      page_limit: 12,
    };

    // this.generateList = this.generateList.bind(this);
    this.searchBar = this.searchBar.bind(this);
    this.search = this.search.bind(this);
  }

  componentDidMount() {
    const query = '?sort=upvotes&order=desc';
    this.props.getRecipes(this.state.page_limit, query).then(() => {
      this.setState({
        All_recipes: this.props.recipes.allRecipes,
      });
    });
  }

  search(e) {
    e.preventDefault();
    const data = { query: this.state.search };
    this.props.searchRecipes(data).then(() => {
      this.setState({
        All_recipes: this.props.recipes.search,
      });
    });
  }
  searchBar(e) {
    this.setState({
      search: e.target.value,
    });
    this.componentDidMount();
  }

  render() {
    const { search } = this.state;
    const headerImg = {
      background: `linear-gradient(90deg, rgba(241, 193, 89, 0.45), rgba(241, 193, 89, 0.5), rgba(241, 193, 89, 0.6), rgba(241, 193, 89, 0.7), rgba(241, 193, 89, 0.8), rgba(241, 193, 89, 0.9), rgba(241, 193, 89, 1)), 
      url('../img/Food.jpg') no-repeat center center`,
      backgroundSize: 'cover',
    };
    return (
      <div>
        <section className="container-fluid ">
          <div className="header" style={headerImg}>
            <Navbar />
            <div className="row header-items justify-content-center">
              <div className="col-lg-6 col-sm-10 text-center">
                <img src="../img/logo.png" alt="logo" />
                <p className=" mt-5 text-dark bg-mirror header-title">
                  “I hate the notion of a secret recipe. Recipes are by nature derivative and meant
                  to be shared that is how they improve, are changed, how new ideas are formed. To
                  stop a recipe in it's tracks, to label it "secret" just seems mean.” ― Molly
                  Wizenberg
                </p>
              </div>
            </div>
          </div>
        </section>
        <section className="container catalog-wrapper" id="catalog">
          <div className="col-12">
            <div className="clearfix">
              <h4 className="float-left fresh-title">Top Recipes Today</h4>
              <h5 className="float-right">
                <Link to="/catalog" className="btn btn-dark hvr-icon-wobble-horizontal">
                  see all recipes{` `}
                </Link>
              </h5>
            </div>
          </div>
          <CatalogList catalog={this.state.All_recipes} />
        </section>
      </div>
    );
  }
}

const mapStateToProps = state => {
  console.log(state.recipes);
  return { recipes: state.recipes };
};

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators(
    {
      getRecipes,
      searchRecipes,
    },
    dispatch,
  ),
});
export default connect(mapStateToProps, mapDispatchToProps)(Home);
