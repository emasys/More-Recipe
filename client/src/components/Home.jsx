import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import DynamicHeader from 'react-sticky-dynamic-header';
import PropTypes from 'prop-types';

// Action
import { getHotRecipes } from '../actions/recipeActions';

//component
import CatalogList from '../components/CatalogList';
import Navbar from './Navbar';
import Auth from './auth';
import BigNavbar from './BigNavbar';

/**
 * Component for Home page
 *
 * @class Home
 * @extends {Component}
 */
export class Home extends Component {
  /**
   * Creates an instance of Home.
   * @param {any} props
   * @memberof Home
   */
  constructor(props) {
    super(props);
    this.state = {
      show: true
    };
  }
  /**
   *
   *
   * @memberof Home
   *
   * @returns {any} react lifecycle method
   */
  componentDidMount() {
    const query = '?sort=views&order=desc';
    this.props.getHotRecipes(12, 0, query);
  }

  /**
   *
   *
   * @returns {any} renders jsx elements
   * @memberof Home
   */
  render() {
    return (
      <div>
        <section className="container-fluid">
          <div className="header">
            <DynamicHeader hasEffect effectDuration={600} useHeadersDifference>
              <BigNavbar />
              <Navbar className="bg-dark fixed-top m-0 p-0" />
            </DynamicHeader>
            <div className="row header-items justify-content-center">
              <div className="col-lg-10 col-sm-10 text-left banner">
                <div
                  className="home-title mt-100"
                  data-aos="fade-up"
                  data-aos-duration="2000"
                >
                  <span className="firstWord">More </span>
                  Recipes
                </div>
                <div
                  className=" text-white bg-mirror pb-30"
                  data-aos="fade-up"
                  data-aos-duration="2000"
                >
                  <p>
                    This is a platform for you to share the awesome and exciting
                    recipe ideas you have invented or learnt.
                  </p>
                  <p>
                    Recipes are by nature derivative and meant to be shared that
                    is how they improve, are changed, how new ideas are formed.
                  </p>
                  <p>Have fun as you share and explore exciting recipes</p>
                  {!Auth.loggedIn() && (
                    <div className="mt-0">
                      <Link
                        to="/signup"
                        className="btn btn-lg bg-orange bold my-5 text-white p-10 signUp-btn"
                      >
                        Signup to Get Started
                      </Link>
                    </div>
                  )}
                  {Auth.loggedIn() && (
                    <div className="">
                      <Link
                        to="/new"
                        className="btn btn-lg bg-orange bold my-5 text-white p-10 signUp-btn"
                      >
                        Add new recipe
                      </Link>
                    </div>
                  )}
                </div>
              </div>
            </div>
          </div>
        </section>

        <section className="container " id="catalog">
          <div className="catalog-wrapper">
            <div className="col-12 ">
              <div className="clearfix" style={{ zIndex: 700 }}>
                <h4 className="float-left header-title">Top Recipes</h4>
                <h5 className="float-right">
                  <Link
                    to="/catalog"
                    className="btn btn-dark hvr-icon-wobble-horizontal"
                  >
                    see all recipes{` `}
                  </Link>
                </h5>
              </div>
            </div>
            <CatalogList
              {...this.props} showDeleteBtn={false}
              catalog={this.props.recipes.hotRecipes}
            />
            <div className="col-12 ">
              <div className="clearfix">
                <h5 className="text-center">
                  <Link
                    to="/catalog"
                    className="btn btn-dark hvr-icon-wobble-horizontal"
                  >
                    View Latest Recipes{` `}
                  </Link>
                </h5>
              </div>
            </div>
          </div>
        </section>
      </div>
    );
  }
}

Home.propTypes = {
  recipes: PropTypes.object,
  getHotRecipes: PropTypes.func
};
const mapDispatchToProps = dispatch => ({
  ...bindActionCreators({ getHotRecipes }, dispatch)
});
const mapStateToProps = state => ({ recipes: state.recipes });
export default connect(mapStateToProps, mapDispatchToProps)(Home);
