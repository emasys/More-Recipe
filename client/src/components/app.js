import React, { Component } from 'react';
// import Script from 'react-load-script';
// import Header from './header';
import Catalog from '../containers/catalog';
import { Link } from 'react-router-dom';
import { getRecipes, searchRecipes } from '../actions';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

//component
import CatalogList from '../components/catalog';
import Navbar from './navbar';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      search: '',
      All_recipes: '',
      page_limit: 12
    };

    // this.generateList = this.generateList.bind(this);
    this.searchBar = this.searchBar.bind(this);
    this.search = this.search.bind(this);
  }

  componentDidMount() {
    const query = '?sort=upvotes&order=desc';
    this.props.getRecipes(this.state.page_limit, query).then(() => {
      this.setState({
        All_recipes: this.props.recipes.allRecipes
      });
    });
  }

  search(e) {
    e.preventDefault();
    const data = { query: this.state.search };
    this.props.searchRecipes(data).then(() => {
      this.setState({
        All_recipes: this.props.recipes.search
      });
    });
  }
  searchBar(e) {
    this.setState({
      search: e.target.value
    });
    this.componentDidMount();
  }
  
  render() {
    const { search } = this.state;
    const headerImg = {
      background: `linear-gradient(90deg, rgba(241, 193, 89, 0.45), rgba(241, 193, 89, 0.5), rgba(241, 193, 89, 0.6), rgba(241, 193, 89, 0.7), rgba(241, 193, 89, 0.8), rgba(241, 193, 89, 0.9), rgba(241, 193, 89, 1)), 
      url('../img/Food.jpg') no-repeat center center`,
      backgroundSize: 'cover'
    };
    return (
      <div>
        <section className="container-fluid ">
          <div className="header" style={headerImg}>
            <Navbar />
            <div className="row header-items justify-content-center">
              <div className="col-lg-6 col-sm-10 text-center">
                <img src="../img/logo.png" alt="logo" />
                <p className=" text-shadowed bolden mt-2">
                  More-Recipes provides a platform for you to share the awesome
                  and exciting recipe ideas you have invented or learnt. feel
                  free to Browse through amazing recipes from around the world,
                  or &nbsp;<Link
                    to="/signup"
                    className="btn btn-dark hvr-icon-buzz-out"
                    role="button"
                  >
                    Sign Up&nbsp;
                  </Link>&nbsp;to post yours.
                </p>
              </div>
            </div>
          </div>
        </section>
        <section className="container catalog-wrapper" id="catalog">
          <div className="col-12">
            <div className="clearfix">
              <h4 className="float-left fresh-title">Top Recipes Today</h4>
              <h5 className="float-right">
                <Link
                  to="/catalog"
                  className="btn btn-dark hvr-icon-wobble-horizontal"
                >
                  see all recipes{` `}
                </Link>
              </h5>
            </div>
          </div>
          <CatalogList catalog={this.state.All_recipes} />
        </section>
      </div>
    );
  }
}

const mapStateToProps = state => {
  console.log(state.recipes);
  return { recipes: state.recipes };
};

const mapDispatchToProps = dispatch => ({
  ...bindActionCreators(
    {
      getRecipes,
      searchRecipes
    },
    dispatch
  )
});
export default connect(mapStateToProps, mapDispatchToProps)(App);
