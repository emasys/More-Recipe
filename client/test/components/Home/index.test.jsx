import React from 'react';
import { mount } from 'enzyme';
import Router from 'react-mock-router';

// Component
import { Home, mapStateToProps } from '../../../src/components/Home';
import fakeStore from '../../__mocks__/fakeStore';

let wrapper = null;

jest.mock('../../../src/components/Navbar');
jest.mock('../../../src/components/BigNavbar');
jest.mock('../../../src/components/CatalogList');
jest.mock('react-sticky-dynamic-header');

describe('Test suite for Home page', () => {
  describe('Render component for unauthenticated user', () => {
    const props = {
      recipes: fakeStore.recipes,
      getHotRecipes: jest.fn(),
      auth: { isLoggedIn: false }
    };
    beforeEach(() => {
      wrapper = mount(<Router>
        <Home {...props} />
      </Router>);
    });

    it('Should render without errors', () => {
      expect(wrapper.length).toEqual(1);
      expect(wrapper.find('#welcome-note').text()).toMatch('This is a platform for you to share the awesome and exciting recipe ideas you have invented or learnt.Recipes are by nature derivative and meant to be shared that is how they improve, are changed, how new ideas are formed.Have fun as you share and explore exciting recipesSignup to Get Started');
    });

    it('should assert for texts on action buttons', () => {
      expect(wrapper.find('a.signUp-btn').text()).toMatch('Signup to Get Started');
      expect(wrapper.find('h4.float-left.header-title').text()).toMatch('Top Recipes');
      expect(wrapper.find('a#view-all-recipes').text()).toMatch('see all recipes');
    });
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should test mapStateToProps', () => {
      const initialState = {
        user: {},
        recipes: {}
      };
      expect(mapStateToProps(initialState).auth).toEqual({});
      expect(mapStateToProps(initialState).recipes).toEqual({});
    });
  });

  describe('Render component for authenticated user', () => {
    const props = {
      recipes: fakeStore.recipes,
      getHotRecipes: jest.fn(),
      auth: fakeStore.user
    };
    beforeEach(() => {
      wrapper = mount(<Router>
        <Home {...props} />
      </Router>);
    });

    it('Should render without errors', () => {
      expect(wrapper.length).toEqual(1);
    });

    it('should assert for texts on action buttons', () => {
      expect(wrapper.find('a.signUp-btn').text()).toMatch('Add new recipe');
      expect(wrapper.find('h4.float-left.header-title').text()).toMatch('Top Recipes');
      expect(wrapper.find('a#view-all-recipes').text()).toMatch('see all recipes');
    });
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
