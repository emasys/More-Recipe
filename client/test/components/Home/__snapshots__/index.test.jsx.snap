// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Home page Render component for authenticated user should match snapshot 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <Home
    auth={
      Object {
        "allUsers": Object {
          "success": true,
          "users": Array [
            Object {
              "avatar": null,
              "bio": null,
              "country": null,
              "createdAt": "2018-03-13T05:04:37.443Z",
              "email": "emasysnd@gmail.com",
              "firstName": null,
              "id": 1,
              "lastName": null,
              "moniker": "admin",
              "password": "$2a$10$47oncNYPerLpBX2BfrUSpeUo4KMIU5dsHkJo3hRUWw1n58K6muoUS",
              "updatedAt": "2018-03-13T05:04:37.443Z",
            },
          ],
        },
        "authInfo": Object {
          "isLoggedIn": true,
          "userId": 1,
          "username": "admin",
        },
        "isLoggedIn": true,
        "message": null,
        "userInfo": Object {
          "data": Object {
            "avatar": null,
            "bio": null,
            "country": "Nigeria",
            "email": "emasysnd@gmail.com",
            "firstName": "emmy",
            "id": 1,
            "lastName": "endy",
            "moniker": "admin",
          },
          "success": true,
        },
        "userProfile": Object {
          "data": Object {
            "avatar": null,
            "bio": null,
            "country": null,
            "email": "emasysnd@gmail.com",
            "firstName": null,
            "id": 1,
            "lastName": null,
            "moniker": "admin",
          },
          "success": true,
        },
      }
    }
    getHotRecipes={[Function]}
    recipes={
      Object {
        "allRecipes": Array [],
        "category": Array [
          Object {
            "category": "Lunch",
            "comments": 0,
            "createdAt": "2018-03-13T05:05:02.553Z",
            "description": "Normal food",
            "direction": "Just do it",
            "downvote": 0,
            "favorite": 0,
            "foodImg": "https://res.cloudinary.com/emasys/image/upload/v1514552968/thumb_image_not_available_dqr7h2.png",
            "id": 1,
            "ingredients": Array [
              "water, salt",
            ],
            "name": "A really long recipe title to make sure the words are truncated",
            "reactionDown": Array [],
            "reactionUp": Array [],
            "searchIng": "water, salt",
            "updatedAt": "2018-03-13T05:12:41.611Z",
            "upvote": 0,
            "userId": 1,
            "views": 0,
          },
        ],
        "count": 1,
        "hotRecipes": Array [
          Object {
            "category": "Lunch",
            "comments": 0,
            "createdAt": "2018-03-13T05:05:02.553Z",
            "description": "Normal food",
            "direction": "Just do it",
            "downvote": 0,
            "favorite": 0,
            "foodImg": "https://res.cloudinary.com/emasys/image/upload/v1514552968/thumb_image_not_available_dqr7h2.png",
            "id": 1,
            "ingredients": Array [
              "water, salt",
            ],
            "name": "A really long recipe title to make sure the words are truncated",
            "reactionDown": Array [],
            "reactionUp": Array [],
            "searchIng": "water, salt",
            "updatedAt": "2018-03-13T05:12:41.611Z",
            "upvote": 0,
            "userId": 1,
            "views": 0,
          },
        ],
        "recipeItem": Object {
          "recipe": Object {
            "category": "Lunch",
            "comments": 0,
            "createdAt": "2018-03-13T05:05:02.553Z",
            "description": "just do it",
            "direction": "just do it",
            "downvote": 0,
            "favorite": 0,
            "favorites": Array [],
            "foodImg": "https://res.cloudinary.com/emasys/image/upload/v1514552968/thumb_image_not_available_dqr7h2.png",
            "id": 1,
            "ingredients": Array [
              "water, salt",
            ],
            "name": "How to cook yam",
            "reactionDown": Array [],
            "reactionUp": Array [],
            "searchIng": "water, salt",
            "updatedAt": "2018-03-14T18:21:16.794Z",
            "upvote": 0,
            "userId": 1,
            "views": 0,
          },
          "success": true,
        },
        "searchResult": Array [],
        "updated": false,
        "userRecipes": Array [],
        "userRecipesCount": 0,
      }
    }
  >
    <div>
      <section
        className="container-fluid"
      >
        <div
          className="header"
        >
          <DynamicHeader
            effectDuration={600}
            hasEffect={true}
            useHeadersDifference={true}
          />
          <div
            className="row header-items justify-content-center"
          >
            <div
              className="col-lg-10 col-sm-10 text-left banner"
            >
              <div
                className="home-title mt-100"
                data-aos="fade-up"
                data-aos-duration="2000"
              >
                <span
                  className="firstWord"
                >
                  More 
                </span>
                Recipes
              </div>
              <div
                className=" text-white bg-mirror pb-30"
                data-aos="fade-up"
                data-aos-duration="2000"
                id="welcome-note"
              >
                <p>
                  This is a platform for you to share the awesome and exciting recipe ideas you have invented or learnt.
                </p>
                <p>
                  Recipes are by nature derivative and meant to be shared that is how they improve, are changed, how new ideas are formed.
                </p>
                <p>
                  Have fun as you share and explore exciting recipes
                </p>
                <div
                  className=""
                >
                  <Link
                    className="btn btn-lg bg-orange bold my-5 text-white p-10 signUp-btn"
                    replace={false}
                    to="/new"
                  >
                    <a
                      className="btn btn-lg bg-orange bold my-5 text-white p-10 signUp-btn"
                      onClick={[Function]}
                    >
                      Add new recipe
                    </a>
                  </Link>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        className="container "
        id="catalog"
      >
        <div
          className="catalog-wrapper"
        >
          <div
            className="col-12 "
          >
            <div
              className="clearfix"
              style={
                Object {
                  "zIndex": 700,
                }
              }
            >
              <h4
                className="float-left header-title"
              >
                Top Recipes
              </h4>
              <h5
                className="float-right"
              >
                <Link
                  className="btn btn-dark hvr-icon-wobble-horizontal"
                  id="view-all-recipes"
                  replace={false}
                  to="/catalog"
                >
                  <a
                    className="btn btn-dark hvr-icon-wobble-horizontal"
                    id="view-all-recipes"
                    onClick={[Function]}
                  >
                    see all recipes
                     
                  </a>
                </Link>
              </h5>
            </div>
          </div>
          <Connect(CatalogList)
            auth={
              Object {
                "allUsers": Object {
                  "success": true,
                  "users": Array [
                    Object {
                      "avatar": null,
                      "bio": null,
                      "country": null,
                      "createdAt": "2018-03-13T05:04:37.443Z",
                      "email": "emasysnd@gmail.com",
                      "firstName": null,
                      "id": 1,
                      "lastName": null,
                      "moniker": "admin",
                      "password": "$2a$10$47oncNYPerLpBX2BfrUSpeUo4KMIU5dsHkJo3hRUWw1n58K6muoUS",
                      "updatedAt": "2018-03-13T05:04:37.443Z",
                    },
                  ],
                },
                "authInfo": Object {
                  "isLoggedIn": true,
                  "userId": 1,
                  "username": "admin",
                },
                "isLoggedIn": true,
                "message": null,
                "userInfo": Object {
                  "data": Object {
                    "avatar": null,
                    "bio": null,
                    "country": "Nigeria",
                    "email": "emasysnd@gmail.com",
                    "firstName": "emmy",
                    "id": 1,
                    "lastName": "endy",
                    "moniker": "admin",
                  },
                  "success": true,
                },
                "userProfile": Object {
                  "data": Object {
                    "avatar": null,
                    "bio": null,
                    "country": null,
                    "email": "emasysnd@gmail.com",
                    "firstName": null,
                    "id": 1,
                    "lastName": null,
                    "moniker": "admin",
                  },
                  "success": true,
                },
              }
            }
            catalog={
              Array [
                Object {
                  "category": "Lunch",
                  "comments": 0,
                  "createdAt": "2018-03-13T05:05:02.553Z",
                  "description": "Normal food",
                  "direction": "Just do it",
                  "downvote": 0,
                  "favorite": 0,
                  "foodImg": "https://res.cloudinary.com/emasys/image/upload/v1514552968/thumb_image_not_available_dqr7h2.png",
                  "id": 1,
                  "ingredients": Array [
                    "water, salt",
                  ],
                  "name": "A really long recipe title to make sure the words are truncated",
                  "reactionDown": Array [],
                  "reactionUp": Array [],
                  "searchIng": "water, salt",
                  "updatedAt": "2018-03-13T05:12:41.611Z",
                  "upvote": 0,
                  "userId": 1,
                  "views": 0,
                },
              ]
            }
            getHotRecipes={[Function]}
            recipes={
              Object {
                "allRecipes": Array [],
                "category": Array [
                  Object {
                    "category": "Lunch",
                    "comments": 0,
                    "createdAt": "2018-03-13T05:05:02.553Z",
                    "description": "Normal food",
                    "direction": "Just do it",
                    "downvote": 0,
                    "favorite": 0,
                    "foodImg": "https://res.cloudinary.com/emasys/image/upload/v1514552968/thumb_image_not_available_dqr7h2.png",
                    "id": 1,
                    "ingredients": Array [
                      "water, salt",
                    ],
                    "name": "A really long recipe title to make sure the words are truncated",
                    "reactionDown": Array [],
                    "reactionUp": Array [],
                    "searchIng": "water, salt",
                    "updatedAt": "2018-03-13T05:12:41.611Z",
                    "upvote": 0,
                    "userId": 1,
                    "views": 0,
                  },
                ],
                "count": 1,
                "hotRecipes": Array [
                  Object {
                    "category": "Lunch",
                    "comments": 0,
                    "createdAt": "2018-03-13T05:05:02.553Z",
                    "description": "Normal food",
                    "direction": "Just do it",
                    "downvote": 0,
                    "favorite": 0,
                    "foodImg": "https://res.cloudinary.com/emasys/image/upload/v1514552968/thumb_image_not_available_dqr7h2.png",
                    "id": 1,
                    "ingredients": Array [
                      "water, salt",
                    ],
                    "name": "A really long recipe title to make sure the words are truncated",
                    "reactionDown": Array [],
                    "reactionUp": Array [],
                    "searchIng": "water, salt",
                    "updatedAt": "2018-03-13T05:12:41.611Z",
                    "upvote": 0,
                    "userId": 1,
                    "views": 0,
                  },
                ],
                "recipeItem": Object {
                  "recipe": Object {
                    "category": "Lunch",
                    "comments": 0,
                    "createdAt": "2018-03-13T05:05:02.553Z",
                    "description": "just do it",
                    "direction": "just do it",
                    "downvote": 0,
                    "favorite": 0,
                    "favorites": Array [],
                    "foodImg": "https://res.cloudinary.com/emasys/image/upload/v1514552968/thumb_image_not_available_dqr7h2.png",
                    "id": 1,
                    "ingredients": Array [
                      "water, salt",
                    ],
                    "name": "How to cook yam",
                    "reactionDown": Array [],
                    "reactionUp": Array [],
                    "searchIng": "water, salt",
                    "updatedAt": "2018-03-14T18:21:16.794Z",
                    "upvote": 0,
                    "userId": 1,
                    "views": 0,
                  },
                  "success": true,
                },
                "searchResult": Array [],
                "updated": false,
                "userRecipes": Array [],
                "userRecipesCount": 0,
              }
            }
            showDeleteBtn={false}
          />
          <div
            className="col-12 "
          >
            <div
              className="clearfix"
            >
              <h5
                className="text-center"
              >
                <Link
                  className="btn btn-dark hvr-icon-wobble-horizontal"
                  replace={false}
                  to="/catalog"
                >
                  <a
                    className="btn btn-dark hvr-icon-wobble-horizontal"
                    onClick={[Function]}
                  >
                    View Latest Recipes
                     
                  </a>
                </Link>
              </h5>
            </div>
          </div>
        </div>
      </section>
    </div>
  </Home>
</MockRouter>
`;

exports[`Test suite for Home page Render component for unauthenticated user should match snapshot 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <Home
    auth={
      Object {
        "isLoggedIn": false,
      }
    }
    getHotRecipes={[Function]}
    recipes={
      Object {
        "allRecipes": Array [],
        "category": Array [
          Object {
            "category": "Lunch",
            "comments": 0,
            "createdAt": "2018-03-13T05:05:02.553Z",
            "description": "Normal food",
            "direction": "Just do it",
            "downvote": 0,
            "favorite": 0,
            "foodImg": "https://res.cloudinary.com/emasys/image/upload/v1514552968/thumb_image_not_available_dqr7h2.png",
            "id": 1,
            "ingredients": Array [
              "water, salt",
            ],
            "name": "A really long recipe title to make sure the words are truncated",
            "reactionDown": Array [],
            "reactionUp": Array [],
            "searchIng": "water, salt",
            "updatedAt": "2018-03-13T05:12:41.611Z",
            "upvote": 0,
            "userId": 1,
            "views": 0,
          },
        ],
        "count": 1,
        "hotRecipes": Array [
          Object {
            "category": "Lunch",
            "comments": 0,
            "createdAt": "2018-03-13T05:05:02.553Z",
            "description": "Normal food",
            "direction": "Just do it",
            "downvote": 0,
            "favorite": 0,
            "foodImg": "https://res.cloudinary.com/emasys/image/upload/v1514552968/thumb_image_not_available_dqr7h2.png",
            "id": 1,
            "ingredients": Array [
              "water, salt",
            ],
            "name": "A really long recipe title to make sure the words are truncated",
            "reactionDown": Array [],
            "reactionUp": Array [],
            "searchIng": "water, salt",
            "updatedAt": "2018-03-13T05:12:41.611Z",
            "upvote": 0,
            "userId": 1,
            "views": 0,
          },
        ],
        "recipeItem": Object {
          "recipe": Object {
            "category": "Lunch",
            "comments": 0,
            "createdAt": "2018-03-13T05:05:02.553Z",
            "description": "just do it",
            "direction": "just do it",
            "downvote": 0,
            "favorite": 0,
            "favorites": Array [],
            "foodImg": "https://res.cloudinary.com/emasys/image/upload/v1514552968/thumb_image_not_available_dqr7h2.png",
            "id": 1,
            "ingredients": Array [
              "water, salt",
            ],
            "name": "How to cook yam",
            "reactionDown": Array [],
            "reactionUp": Array [],
            "searchIng": "water, salt",
            "updatedAt": "2018-03-14T18:21:16.794Z",
            "upvote": 0,
            "userId": 1,
            "views": 0,
          },
          "success": true,
        },
        "searchResult": Array [],
        "updated": false,
        "userRecipes": Array [],
        "userRecipesCount": 0,
      }
    }
  >
    <div>
      <section
        className="container-fluid"
      >
        <div
          className="header"
        >
          <DynamicHeader
            effectDuration={600}
            hasEffect={true}
            useHeadersDifference={true}
          />
          <div
            className="row header-items justify-content-center"
          >
            <div
              className="col-lg-10 col-sm-10 text-left banner"
            >
              <div
                className="home-title mt-100"
                data-aos="fade-up"
                data-aos-duration="2000"
              >
                <span
                  className="firstWord"
                >
                  More 
                </span>
                Recipes
              </div>
              <div
                className=" text-white bg-mirror pb-30"
                data-aos="fade-up"
                data-aos-duration="2000"
                id="welcome-note"
              >
                <p>
                  This is a platform for you to share the awesome and exciting recipe ideas you have invented or learnt.
                </p>
                <p>
                  Recipes are by nature derivative and meant to be shared that is how they improve, are changed, how new ideas are formed.
                </p>
                <p>
                  Have fun as you share and explore exciting recipes
                </p>
                <div
                  className="mt-0"
                >
                  <Link
                    className="btn btn-lg bg-orange bold my-5 text-white p-10 signUp-btn"
                    replace={false}
                    to="/signup"
                  >
                    <a
                      className="btn btn-lg bg-orange bold my-5 text-white p-10 signUp-btn"
                      onClick={[Function]}
                    >
                      Signup to Get Started
                    </a>
                  </Link>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        className="container "
        id="catalog"
      >
        <div
          className="catalog-wrapper"
        >
          <div
            className="col-12 "
          >
            <div
              className="clearfix"
              style={
                Object {
                  "zIndex": 700,
                }
              }
            >
              <h4
                className="float-left header-title"
              >
                Top Recipes
              </h4>
              <h5
                className="float-right"
              >
                <Link
                  className="btn btn-dark hvr-icon-wobble-horizontal"
                  id="view-all-recipes"
                  replace={false}
                  to="/catalog"
                >
                  <a
                    className="btn btn-dark hvr-icon-wobble-horizontal"
                    id="view-all-recipes"
                    onClick={[Function]}
                  >
                    see all recipes
                     
                  </a>
                </Link>
              </h5>
            </div>
          </div>
          <Connect(CatalogList)
            auth={
              Object {
                "isLoggedIn": false,
              }
            }
            catalog={
              Array [
                Object {
                  "category": "Lunch",
                  "comments": 0,
                  "createdAt": "2018-03-13T05:05:02.553Z",
                  "description": "Normal food",
                  "direction": "Just do it",
                  "downvote": 0,
                  "favorite": 0,
                  "foodImg": "https://res.cloudinary.com/emasys/image/upload/v1514552968/thumb_image_not_available_dqr7h2.png",
                  "id": 1,
                  "ingredients": Array [
                    "water, salt",
                  ],
                  "name": "A really long recipe title to make sure the words are truncated",
                  "reactionDown": Array [],
                  "reactionUp": Array [],
                  "searchIng": "water, salt",
                  "updatedAt": "2018-03-13T05:12:41.611Z",
                  "upvote": 0,
                  "userId": 1,
                  "views": 0,
                },
              ]
            }
            getHotRecipes={[Function]}
            recipes={
              Object {
                "allRecipes": Array [],
                "category": Array [
                  Object {
                    "category": "Lunch",
                    "comments": 0,
                    "createdAt": "2018-03-13T05:05:02.553Z",
                    "description": "Normal food",
                    "direction": "Just do it",
                    "downvote": 0,
                    "favorite": 0,
                    "foodImg": "https://res.cloudinary.com/emasys/image/upload/v1514552968/thumb_image_not_available_dqr7h2.png",
                    "id": 1,
                    "ingredients": Array [
                      "water, salt",
                    ],
                    "name": "A really long recipe title to make sure the words are truncated",
                    "reactionDown": Array [],
                    "reactionUp": Array [],
                    "searchIng": "water, salt",
                    "updatedAt": "2018-03-13T05:12:41.611Z",
                    "upvote": 0,
                    "userId": 1,
                    "views": 0,
                  },
                ],
                "count": 1,
                "hotRecipes": Array [
                  Object {
                    "category": "Lunch",
                    "comments": 0,
                    "createdAt": "2018-03-13T05:05:02.553Z",
                    "description": "Normal food",
                    "direction": "Just do it",
                    "downvote": 0,
                    "favorite": 0,
                    "foodImg": "https://res.cloudinary.com/emasys/image/upload/v1514552968/thumb_image_not_available_dqr7h2.png",
                    "id": 1,
                    "ingredients": Array [
                      "water, salt",
                    ],
                    "name": "A really long recipe title to make sure the words are truncated",
                    "reactionDown": Array [],
                    "reactionUp": Array [],
                    "searchIng": "water, salt",
                    "updatedAt": "2018-03-13T05:12:41.611Z",
                    "upvote": 0,
                    "userId": 1,
                    "views": 0,
                  },
                ],
                "recipeItem": Object {
                  "recipe": Object {
                    "category": "Lunch",
                    "comments": 0,
                    "createdAt": "2018-03-13T05:05:02.553Z",
                    "description": "just do it",
                    "direction": "just do it",
                    "downvote": 0,
                    "favorite": 0,
                    "favorites": Array [],
                    "foodImg": "https://res.cloudinary.com/emasys/image/upload/v1514552968/thumb_image_not_available_dqr7h2.png",
                    "id": 1,
                    "ingredients": Array [
                      "water, salt",
                    ],
                    "name": "How to cook yam",
                    "reactionDown": Array [],
                    "reactionUp": Array [],
                    "searchIng": "water, salt",
                    "updatedAt": "2018-03-14T18:21:16.794Z",
                    "upvote": 0,
                    "userId": 1,
                    "views": 0,
                  },
                  "success": true,
                },
                "searchResult": Array [],
                "updated": false,
                "userRecipes": Array [],
                "userRecipesCount": 0,
              }
            }
            showDeleteBtn={false}
          />
          <div
            className="col-12 "
          >
            <div
              className="clearfix"
            >
              <h5
                className="text-center"
              >
                <Link
                  className="btn btn-dark hvr-icon-wobble-horizontal"
                  replace={false}
                  to="/catalog"
                >
                  <a
                    className="btn btn-dark hvr-icon-wobble-horizontal"
                    onClick={[Function]}
                  >
                    View Latest Recipes
                     
                  </a>
                </Link>
              </h5>
            </div>
          </div>
        </div>
      </section>
    </div>
  </Home>
</MockRouter>
`;
